### ISSUE TITLE
[chef-client][client.rb] - Setting node_name

### ISSUE THREAD

**Issue by username-is-already-taken2** [chef-cookbooks/chef-client#493]
### Cookbook version
8.0.1

### Chef-client version
12.19.33

### Platform Details
Windows

### Scenario:
On windows the hostname name of the machine changes after its joined to the domain, we register the node with chef but then becuases disconnected after joining the domain.

Up until this point we haven't hardcoded the `node_name` attribute within the nodes client.rb the default chef-client::config cookbook leaves us with `# Using default nodes name (fqdn)` which has worked fine.

I've been trying to use the `chef-client::config` recipe to amend the client.rb and add the node_name attribute but I can't get it to add it using an attribute ie

if I set this attribute

`node['chef_client']['config']['node_name'] = "#{node['machinename']}`

the `client` recipe literally writes this to the `client.rb` file and I end up with this
```
node_name "\#{node['machinename']}"
```

Looking in the template, I'm not familar with the .inspect method is that whats stopping it from translating it?
```
<% if @chef_config['node_name'] -%>
node_name <%= @chef_config['node_name'].inspect %>
<% else -%>
# Using default node name (fqdn)
<% end -%>
```

**Comment by cheeseplus** [comment]
You wouldn't want to ammend the node_name after a node has already been bootstrapped as node_name is going to be used on registration and is then set in stone (relatively).

**Comment by krzkowalczyk** [comment]
Hi,
We've encounter the same problem with changing node name after joining the domain.
The solution which we have chosen it's to force node name by setting a normal attribute with node name during bootstrap process. 
```
      "chef_client": {
        "config": {
          "node_name": "mynodename"
        }
      }
```

**Comment by tas50** [comment]
Fixed in 10.0
